### PR TITLE
contrib/oath-toolkit: new package (2.6.11)

### PR DESCRIPTION
**Pull request by jcgruenhage**
Currently not working, failing at https://gitlab.com/oath-toolkit/oath-toolkit/-/blob/master/libpskc/schemas/Makefile.am?ref_type=heads#L27-L36 and I have no clue how to fix that.

### CODE REVIEW

**Comment by nekopsykose** [comment]
xmlcatalog is in libxml2-progs, not libxml2 for hostmakedepends

**Comment by nekopsykose** [comment]
after that it gets further but fails in ./gdoc because that has no shebang seemingly, not sure about that

File: contrib/oath-toolkit/template.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+pkgname = "oath-toolkit"
+pkgver = "2.6.11"
+pkgrel = 0
+build_style = "gnu_configure"
+make_cmd = "gmake"
+make_dir = "."
```
Comment (jcgruenhage) [inline contrib/oath-toolkit/template.py:6]: @nekopsykose ^ this bit fixes the `./gdoc` issue. Building out of tree isn't supported it seems

**Comment by jcgruenhage** [comment]
I haven't investigated the check failures further, but at least `oathtool` works when used in `pass-otp`.

File: contrib/oath-toolkit/template.py (hunk 1 above)
Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:6]: nice :)

File: contrib/oath-toolkit/template.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+url = "http://www.nongnu.org/oath-toolkit"
+source = f"http://download.savannah.gnu.org/releases/{pkgname}/{pkgname}-{pkgver}.tar.gz"
+sha256 = "fc512a4a5b46f4c43ab0586c3189fece4d54f7e649397d6fa1e23428431e2cb4"
+# TODO: failing checks are yet to be investigated
+options = ["!check"]
+
+
+@subpackage("oathtool")
+def _oathtool(self):
+    return ["usr/bin/oathtool"]
```
Comment (nekopsykose) [inline contrib/oath-toolkit/template.py:28]: this also has to take the manpage

Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py]: done

File: contrib/oath-toolkit/template.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+build_style = "gnu_configure"
+make_cmd = "gmake"
+make_dir = "."
+hostmakedepends = [
+    "gmake",
+    "automake",
+    "libtool",
+    "gtk-doc-tools",
+    "pkgconf",
+    "libxml2-progs",
```
Comment (nekopsykose) [inline contrib/oath-toolkit/template.py:13]: sort

File: contrib/oath-toolkit/template.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+source = f"http://download.savannah.gnu.org/releases/{pkgname}/{pkgname}-{pkgver}.tar.gz"
+sha256 = "fc512a4a5b46f4c43ab0586c3189fece4d54f7e649397d6fa1e23428431e2cb4"
+# TODO: failing checks are yet to be investigated
+options = ["!check"]
+
+
+def post_install(self):
+    for lib in ["pskc", "oath"]:
+        self.install_man(f"lib{lib}/man/{lib}_*.3", glob=True)
+        self.install_man(f"{lib}tool/{lib}tool.1")
```
Comment (nekopsykose) [inline contrib/oath-toolkit/template.py:29]: is there a reason the build system doesn't install these?

File: contrib/oath-toolkit/template.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+
+@subpackage("oathtool")
+def _oathtool(self):
+    return ["usr/bin/oathtool", "usr/share/man/man1/oathtool.1"]
+
+
+@subpackage("pskctool")
+def _pskctool(self):
+    return ["usr/bin/pskctool", "usr/share/man/man1/pskctool.1"]
```
Comment (nekopsykose) [inline contrib/oath-toolkit/template.py:39]: hmm.. thinking again, is there a need to split these? if the whole thing is like just a few mb then there's usually not much point

File: contrib/oath-toolkit/template.py (hunk 4 above)
Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py:29]: I'm not sure why it doesn't, but it doesn't. Alpine for example also doesn't ship the man pages.

File: contrib/oath-toolkit/template.py (hunk 5 above)
Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py:39]: I thought it was good practice to split them, but I have no problem merging them either.

Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:39]: splitting is generally to split up large stuff, or things that pull in some big dep chain (e.g. maybe some gui component), or auto policy stuff (default splits, devel , etc). but for small programs like this it's usually more hassle (entire thing is very small overall and having to hunt for a few binaries instead of installing just 1 thing is usually an extra annoyance).

so compared to e.g. alpine there is less granular splitting here per-program or whatever as it's not as final-size focused

Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py:39]: So I should get rid of this split and just keep the devel subpackage?

File: contrib/oath-toolkit/template.py (hunk 4 above)
Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py:29]: Is there anything I should do regarding this? Or is it also okay to keep it like this?

Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:29]: the same is done for the .3's?

Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py:29]: I called `apk manifest` on the packages and couldn't find the man pages, maybe I looked in the wrong place. Let me try again.

Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:29]: yeah i removed this post_install and the same manpages are already installed

Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:29]: alpine also has them https://pkgs.alpinelinux.org/contents?branch=edge&repo=community&arch=x86_64&page=2&name=oath%2dtoolkit%2ddoc

Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:29]: > I called apk manifest on the packages and couldn't find the man pages,  

they are split to -man by default. for .3, it's -devel-man. in both cases, they autoinstall with some base- package when the respective one is installed

File: contrib/oath-toolkit/template.py (hunk 5 above)
Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:39]: yeah

File: contrib/oath-toolkit/template.py (hunk 4 above)
Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py:29]: yeah okay I apparently just checked the wrong packages, because the man pages are there.

File: contrib/oath-toolkit/template.py (hunk 5 above)
Comment (jcgruenhage) [reply contrib/oath-toolkit/template.py:39]: So, `oath-toolkit-progs` or should they just go into the main package?

Comment (nekopsykose) [reply contrib/oath-toolkit/template.py:39]: if they're 45kb then there's no point to split that imo
